rmf_task/src/rmf_task/Priority.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef SRC__RMF_TASK__PRIORITY_HPP
+#define SRC__RMF_TASK__PRIORITY_HPP
+
+#include <memory>
+
+#include <rmf_utils/impl_ptr.hpp>
+
+namespace rmf_task {
+
+/// Implement this for a new prioritization scheme.
+class Priority
+{
+public:
+
+  /// Get the value of this priority
+  virtual std::size_t value() const = 0;
+
+  virtual ~Priority() = default;
+};
+
+using PriorityPtr = std::shared_ptr<Priority>;
+using ConstPriorityPtr = std::shared_ptr<const Priority>;
+
+//==============================================================================
+// Sample implementation for binary prioritization scheme
+class BinaryPriority : public Priority
```

Comment:
Let's define `BinaryPriority` in its own header file for the sake of cleaner code organization.

Reply:
Done https://github.com/osrf/rmf_core/pull/278/commits/b7d22e17ef093fa6fd62d77b62ab5353f24c1630